Updates and corrections to wire protocol docs (#7)
canonical/dqlite-docs: reference/wire-protocol.md

## Setup

> As soon as a connection is established, the client must send to the server a single word containing the protocol version it wishes to use. The current protocol version is 1 (in little-endian representation).

Reviewer: Would it be easier to just give the actual word?
```suggestion
As soon as a connection is established, the client must send to the server a single word containing the protocol version it wishes to use. The current protocol version is 1 (`00000001`).
```
(If that one is correct :smile: )

Reviewer: Nah, not really - let's leave it as is. :)

## Conversation

> After the setup, communication between client and server happens by message exchange. Typically the client will send to the server a message containing a request and the server will send to the client a message containing a response. Any request may result in an failure response (type `0`); otherwise, the correspondence between request and response types is as follows:

Reviewer: ```suggestion
After the setup, communication between client and server happens by message exchange. Typically, the client will send to the server a message containing a request and the server will send to the client a message containing a response. Any request may result in a failure response (type `0`); otherwise, the correspondence between request and response types is as follows:
```

| Request | Response |
| ---     | ---      |
| `0` (Get current leader) | `1` (Leader information) |
| `1` (Client registration) | `2` (Welcome) |
| `3` (Open a database) | `4` (Database information) |
| `4` (Prepare a statement) | `5` (Prepared statement information) |
| `5` (Execute a prepared statement)` | `6` (Statement execution result) |
| `6` (Execute a prepared statement yielding rows) | `7` (Batch of table rows) |

> | `7` (Finalize a prepared statement) | `8` (Acknowledgement) |

Reviewer: ```suggestion
| `7` (Finalise a prepared statement) | `8` (Acknowledgement) |
```
(Unless that is a fixed term, of course.)

| `8` (Execute a SQL text) | `6` (Statement execution result) |

> | `9` (Execute a SQL text yielding rows) | `7` (Batch of table rows) | `10` (Interrupt the execution of a statement yielding rows) | `8` (Acknowledgement)

Reviewer: Missing a new line here?
